### PR TITLE
Update to Akka 2.5.31 and add ClassicActorSystemProvider in extensions, #298

### PR DESCRIPTION
**Pull request by patriknw**
References #298

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -7,7 +7,7 @@ object Dependencies {
   val Scala212 = "2.12.10"
   val Scala213 = "2.13.0"
 
-  val AkkaVersion = "2.5.27"
+  val AkkaVersion = "2.5.31"
```
Comment (patriknw) [inline project/Dependencies.scala:10]: it was backported in https://github.com/akka/akka/pull/28765

**Comment by chbatey** [comment]
Need to update the maven integration test Akka Version

**Comment by chbatey** [comment]
I can do that now

**Comment by chbatey** [comment]
Manually checked that Travis has passed, not sure why it isn't reporting as green

**Pull request merged**